libs/blocks/preflight/checks/preflightApi.js

```diff
@@ -53,3 +55,46 @@ export default {
     runChecks: runChecksSeo,
   },
 };
+
+let preflightResults;
+
+export async function getPreflightResults(url, area, useCache = true) {
+  if (useCache && checks) {
+    preflightResults = await checks;
+    const returnValue = {
+      isViewportTooSmall: isViewportTooSmall(),
+      runChecks: preflightResults,
+    };
+    return returnValue;
+  }
+
+  checks = (async () => {
+    const assets = await Promise.all(runChecksAssets(url, area));
+    const performance = await Promise.all(runChecksPerformance(url, area));
+    const seo = await Promise.all(runChecksSeo({ url, area }));
+    return {
+      assets,
+      performance,
+      seo,
+    };
+  })();
+
+  preflightResults = await checks;
+
+  return {
+    isViewportTooSmall: isViewportTooSmall(),
+    runChecks: preflightResults,
+  };
+}
```

Comment:
Formatting is horrible, but I'd probably do it like this:
```suggestion
const runChecks = async () => {
      const assets = await Promise.all(runChecksAssets(url, area));
      const performance = await Promise.all(runChecksPerformance(url, area));
      const seo = await Promise.all(runChecksSeo({ url, area }));
      return {assets, performance, seo}
}

export async function getPreflightResults(url, area, useCache = true) {
  if (useCache) {
cachedChecks = checks || await runChecks()
  return {
    isViewportTooSmall: isViewportTooSmall(),
    runChecks: cachedChecks,
  };
  }
   const res = await runChecks()
    return {
      isViewportTooSmall: isViewportTooSmall(),
      runChecks: res,
    }
}
```